### DISCUSSION TITLE
Seurat V5 - var.features

### DISCUSSION THREAD

**Discussion by Dooo0k** [satijalab/seurat#8213] [Q&A]
hello.

I am currently trying to check hierarchical clustering, but I have confirmed that var.features data does not exist separately in Seurat V5.
Then, I wonder where the data corresponding to var.features is stored in Seurat V5.

thank you

**Comment by mluciarr** [comment]
Hi @Dooo0k. 
I got the same issue and I'm struggling to create the `var.feature` variable inside the object. 
Did you eventually get a solution for that? 
Or  I'd really appreciate it if someone could help us.
Thanks a lot.

Lucia

**Reply by Dooo0k** [reply]
Hi, @mluciarr .

Variable features appear to be stored in obj@assays$RNA@layers$scale.data.
I solved it with the command above, and I think it would be better to check if it is the same in your data.

I hope it helps you. 
Thank you

**Reply by mluciarr** [reply]
Hi @Dooo0k 

I'm unsure if "scale.data" is equivalent to "var.features". I'd say that they are distinct. Scaling data and selecting variable features seem to be separate processes. Also, my concern arises from the necessity of having the "var.features" slot in my object to execute MOFA (a specific package). Therefore, I need to find a way to generate it in my seurat object.

In any case, I really appreciate your recommendation and help! 👍🏼 :)

**Reply by mluciarr** [reply]
I think the var.features in Seurat v5 is in SeuratObj@assays$RNA@meta.data$var.features
but I don't know how to take it out of the metadata compartment.

**Reply by Dooo0k** [reply]
Hi, @mluciarr 
Thank you for the reply.

I was also unsure if 'scale.data' was considered a Variable features. But as far as I know, the 'ScaleData' function is usually used for genes that were selected in the 'Find variable features' step.

I think you can extract genes from the 'obj@assays$RNA@features@.Data' command by filtering for those with a 'TRUE' value for 'scale.data'.

**Reply by Dooo0k** [reply]
Furthermore, if you computed variable genes during PCA calculation, you can review the list of genes specified for PCA calculation through 'obj@reductions[["pca"]]@feature.loadings'.

If my understanding is incorrect, please provide feedback :D

**Reply by apal6** [reply]
Hi, I would like to follow up on this thread. I have the same issue. I encountered the error when I tried to project my data on a referEnce data (ProjectTil package). Error: 
`Error: BiocParallel errors
  1 remote errors, element index: 1
  0 unevaluated and other errors
  first remote error:
Error in as.vector(x): no slot of name "var.features" for this object of class "Assay5"
`
Would appreciate if someone could help? 
Thank you in advance!

**Reply by amrita-singhs** [reply]
Haven't managed to find the slot in the object yet, but I'm using "VariableFeatures(object)" to access the list. Hope this helps someone
